File: node/packages/content/src/services/SummarizationService.ts
export class SummarizationService {
    static async summarize(text: string, authHeaders: Record<string, string>): Promise<string> {
      const response = await fetch(`${process.env.API_URL}/llm/predict`, {
        method: "POST",
        headers: { "Content-Type": "application/json", ...authHeaders },
        body: JSON.stringify({
          uri: "/v1/chat/completions",
          provider: "chatgpt",
          model: "gpt-5-nano",
          payload: {
            messages: [
              { role: "system", content: "Summarize this text into 2–3 clear sentences for study purposes." },
              { role: "user", content: text }
            ]
          }
        })
      });
  
      if (!response.ok) return "";
      const data = await response.json();
      return data?.choices?.[0]?.message?.content?.trim() ?? "";
    }
  
    static async generate(prompt: string, authHeaders: Record<string, string>): Promise<string> {
      const response = await fetch(`${process.env.API_URL}/llm/predict`, {
        method: "POST",
        headers: { "Content-Type": "application/json", ...authHeaders },
        body: JSON.stringify({
          uri: "/v1/chat/completions",
          provider: "chatgpt",
          model: "gpt-5-nano",
          payload: { messages: [{ role: "user", content: prompt }] }
        })
      });
  
      if (!response.ok) return "";
      const data = await response.json();
      return data?.choices?.[0]?.message?.content?.trim() ?? "";
    }
  }
  

File: python/packages/package-unit/src/package_unit/services/summarization_service.py
from __future__ import annotations

from typing import Dict

from package_unit.utils.http import post_json


class SummarizationService:
    @staticmethod
    def summarize(text: str, auth_headers: Dict[str, str]) -> str:
        resp = post_json(
            "llm/predict",
            json={
                "uri": "/v1/chat/completions",
                "provider": "chatgpt",
                "model": "gpt-5-nano",
                "payload": {
                    "messages": [
                        {
                            "role": "system",
                            "content": "Summarize this text into 2–3 clear sentences for study purposes.",
                        },
                        {"role": "user", "content": text},
                    ]
                },
            },
            headers={"Content-Type": "application/json", **(auth_headers or {})},
        )
        if not resp.ok:
            return ""
        try:
            data = resp.json()
            return (
                (data.get("choices") or [{}])[0]
                .get("message", {})
                .get("content", "")
                .strip()
            )
        except Exception:
            return ""

    @staticmethod
    def generate(prompt: str, auth_headers: Dict[str, str]) -> str:
        resp = post_json(
            "llm/predict",
            json={
                "uri": "/v1/chat/completions",
                "provider": "chatgpt",
                "model": "gpt-5-nano",
                "payload": {"messages": [{"role": "user", "content": prompt}]},
            },
            headers={"Content-Type": "application/json", **(auth_headers or {})},
        )
        if not resp.ok:
            return ""
        try:
            data = resp.json()
            return (
                (data.get("choices") or [{}])[0]
                .get("message", {})
                .get("content", "")
                .strip()
            )
        except Exception:
            return ""
